tools/sql_extraction/src/main/kotlin/DataFlowEngine.kt

```diff
@@ -30,6 +30,25 @@ class DataFlowEngine(private val environment: Environment = Environment()) {
             .map { QueryUsages(it.key, it.value.sorted()) }
     }
 
+    /**
+     * Visits an annotation located at [location].
+     * Strings passed as arguments can be considered a usage.
+     *
+     * @param[visitArguments] Potential visit for all annotation arguments.
+     */
+    fun visitAnnotation(location: Location, visitArguments: () -> Unit) {
+        val prevQuery = currentQuery
+
+        currentQuery = null
+        visitArguments()
+
+        if (currentQuery != null) {
```

Comment:
For the sample test 
```
   @SqlUpdate("insert into users (id, name) values (?, ?)")
    void insert(long id, String name);
```

What is the value of `currentQuery` here?

Reply:
It would be the QueryFragment corresponding to the following json:
```javascript
{
  "count": "SINGLE",
  "location": {
    "startLine": 1,
    "startColumn": 17, // index of 'i' right after the first quote
    "endLine": 1,
    "endColumn": 58 // index of ')' right before the second quote
  },
  "literal": "insert into users (id, name) values (?, ?)" // quotes are not included
}
```

Reply:
`CurrentQuery` will only be null if no string literals or variables holding query fragments are found in the annotation arguments.